crates/oracle/src/runner/oracle.rs

```diff
@@ -205,8 +205,8 @@ impl Oracle {
         info!("Owner ETH Balance is {} gwei", balance);
 
         // overflow check
-        if balance.bits() as u32 > i64::BITS / 2 {
-            // number can't be represented as i64
+        if balance.bits() as u32 > i64::BITS - 1 {
+            // number can't be represented as positive i64
             METRICS.set_wallet_balance(i64::MAX);
         } else {
             METRICS.set_wallet_balance(balance.as_u64() as i64);
```

Comment:
How about...

```rust
// overflow check
if let Ok(balance) = i64::try_from(balance) {
  METRICS.set_wallet_balance(balance);
} else {
  // The number can't be represented as a positive `i64`, we set it to
  // the max value.
  METRICS.set_wallet_balance(i64::MAX);
};
```

You can also do `METRICS.set_wallet_balance(i64::try_from(balance).unwrap_or(i64::MAX));` if you prefer.

It works the same, the conversion attempt is explicit with `TryFrom`, and we avoid using [`as`](https://users.rust-lang.org/t/as-considered-harmful/35338).

Reply:
Thanks for the suggestion! Implemented in b2abdab